components/esb-tools/plugins/org.wso2.integrationstudio.esb.form.editors/src/org/wso2/integrationstudio/esb/form/editors/article/providers/ValueExpressionWidgetNameSpacedPropertyEditorDialog.java

```diff
@@ -0,0 +1,1746 @@
+                                    final Node rootNode = document.getDocumentElement();
+
+                                    // Adding a tree item is a ui related
+                                    // operation which must be executed on the
+                                    // ui thread. Since we are currently
+                                    // executing inside a background thread, we
+                                    // need to explicitly invoke the ui thread.
+                                    new UIJob(UI_JOB_ID) {
+                                        public IStatus runInUIThread(IProgressMonitor monitor) {
+                                            addTreeItem(null, rootNode);
+                                            addNameSpaces(rootNode);
+                                            collectedNamespaces.clear();
+                                            nsListBox.removeAll();
+                                            for (Entry<String, String> nsEntry : getNameSpaces().entrySet()) {
+                                                addNamespace(nsEntry.getKey(), nsEntry.getValue());
+                                            }
+                                            
+                                            // Get formatted XML content
+                                            final String xmlFileContentStr = xPathValidator.getFormattedXMLStringFromDoc(document);
+                                            inlineXMLTextArea.setText(xmlFileContentStr);
+                                            infoLabel.setText(TREE_VIEW_INFO_LABEL_TEXT);
+                                            isCurrentXMLContentValid = true;
+                                            isInlineXMLTextAreaDirty = true;
+                                            
+                                            return Status.OK_STATUS;
+                                        }
+                                    }.schedule();
+
+                                    // Done.
+                                    monitor.done();
+                                } catch (Exception ex) {
+                                    // Stop progress.
+                                    monitor.done();
+
+                                    // Report error.
+                                    throw new InvocationTargetException(ex);
+                                }
+                            }
+                        };
+
+                        // Run the operation within a progress monitor dialog,
+                        // make sure to fork-off from the ui thread so that we
+                        // don't cause the ui to hang.
+                        try {
+                            new ProgressMonitorDialog(dialogShell).run(true, false, runnable);
+                        } catch (Exception ex) {
+                            MessageDialog.openError(dialogShell, "Parsing Error",
```

Comment:
Don't we have to pass the original exception?

Reply:
Fixed in https://github.com/wso2/integration-studio/pull/1114